main.tf

```diff
@@ -53,14 +53,15 @@ resource "ibm_resource_key" "resource_key" {
 
 locals {
   cos_instance_id      = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].id)[0] : var.existing_cos_instance_id
+  cos_instance_guid    = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].guid)[0] : element(split(":", var.existing_cos_instance_id), length(split(":", var.existing_cos_instance_id)) - 3)
```

Comment:
The original was 
```hcl
cos_instance_guid = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].guid)[0] : tolist(data.ibm_resource_instance.cos_instance[*].guid)[0]
```
It looks like you are editing the instance id, to match the guid. Is this a bit fragile? I do like not having the data lookup but which is better/more stable?

Reply:
It does not take a cos instance id